src/Microsoft.AspNetCore.ResponseCaching/Internal/ResponseCachingKeyProvider.cs

```diff
@@ -13,14 +13,31 @@ namespace Microsoft.AspNetCore.ResponseCaching.Internal
 {
     public class ResponseCachingKeyProvider : IResponseCachingKeyProvider
     {
+        private const string UseMethodPathCacheKeySwitch = "Switch.Microsoft.AspNetCore.ResponseCaching.UseMethodPathCacheKey";
+
         // Use the record separator for delimiting components of the cache key to avoid possible collisions
         private static readonly char KeyDelimiter = '\x1e';
 
         private readonly ObjectPool<StringBuilder> _builderPool;
         private readonly ResponseCachingOptions _options;
 
+        internal bool UseMethodPathCacheKey { get; set; }
+
         public ResponseCachingKeyProvider(ObjectPoolProvider poolProvider, IOptions<ResponseCachingOptions> options)
         {
+#if NET451
+            var value = System.Configuration.ConfigurationManager.AppSettings.GetValues(UseMethodPathCacheKeySwitch)?.FirstOrDefault();
+            var isEnabled = false;
+            if (bool.TryParse(value, out isEnabled))
+            {
+                UseMethodPathCacheKey = isEnabled;
+            }
+#else
+            var isEnabled = false;
+            AppContext.TryGetSwitch(UseMethodPathCacheKeySwitch, out isEnabled);
```

Comment:
Can you can use UseMethodPathCacheKey directly?

Reply:
I tried that and hit `error CS0206: A property or indexer may not be passed as an out or ref parameter`